app/app/BaseUser.php

```diff
@@ -283,6 +283,26 @@ public static function confirmEmail(string $email,string $confirmCode)
 
 		return false;
 	}
+	/**
+	* send password recovery verification email to user's mailbox
+	*
+	*/
+	public static function sendPasswordRecoveryEmail($user)
+	{
+
+		$password_recovery_token = str_random(60); //generate email verification token
```

Comment:
60 or more seems like a secure enough length but the length should match the value in 2016_10_30_053317_create_initial_database_schema.php.  If you use 60 here, replace 255 with 60 in the schema.